src/utils/session.ts

```diff
@@ -0,0 +1,57 @@
+import Peer from "peerjs";
+import { batch, computed, effect, signal } from "@preact/signals";
+import { connectionMap, peer, PEER_ID_PREFIX, PEER_JS_OPTIONS } from "./peer";
+
+export const hostId = computed(() => `${PEER_ID_PREFIX}-${roomName}`);
+export const roomName = signal<string>();
+export const playerName = signal<string>();
+export const playerMap = signal<Map<string, string>>(new Map());
+
+effect(() => {
+  if (!peer.value) {
+    exitRoom();
+    return;
+  }
+});
+
+export function createRoom() {
+  if (!roomName.value) {
+    alert("A room name is required.");
+    return;
+  }
+  if (!playerName.value) {
+    alert("A player name is required.");
+    return;
+  }
+  peer.value?.destroy();
+  peer.value = new Peer(hostId.value, PEER_JS_OPTIONS);
+  console.info(
+    `Create a room with name ${roomName} as ${playerName} and wait for players to join.`
+  );
+  playerMap.value = new Map([[peer.value.id, playerName.value]]);
+}
+
+export function joinRoom() {
+  // TODO should prevent player to join a room that is not yet created
```

Comment:
Can we create an issue and link it here?

Reply:
I guess this would need to be solved in a later pr, this refactoring pr is already much larger than expected.